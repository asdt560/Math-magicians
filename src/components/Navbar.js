import React, { useEffect } from 'react';
import { Link } from 'react-router-dom';

function Navbar() {
  useEffect(() => {
    document.title = 'Calculator';
  });
  return (
    <header>
      <h1>Math Magicians</h1>
      <ul>
        <li className="navitem"><Link to="/">Home</Link></li>
        <li className="navitem"><Link to="/calculator">Calculator</Link></li>
        <li className="navitem"><Link to="/quote">Quote</Link></li>
      </ul>
    </header>
  );
}

export default Navbar;
