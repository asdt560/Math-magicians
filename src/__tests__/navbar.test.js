import '@testing-library/jest-dom';
import React from 'react';
import { render } from '@testing-library/react';
import { BrowserRouter as Router } from 'react-router-dom';
import Navbar from '../components/Navbar';

it('Navbar component', async () => {
  render(
    <Router>
      <Navbar />
    </Router>,
  );
  const navbar = `<header>
    <h1>Math Magicians</h1>
    <ul>
      <li className="navitem"><Link to="/">Home</Link></li>
      <li className="navitem"><Link to="/calculator">Calculator</Link></li>
      <li className="navitem"><Link to="/quote">Quote</Link></li>
    </ul>
  </header>`;
  expect(navbar).toMatchSnapshot();
});
