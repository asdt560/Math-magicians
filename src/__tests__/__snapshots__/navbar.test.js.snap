// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Navbar component 1`] = `
"<header>
    <h1>Math Magicians</h1>
    <ul>
      <li className=\\"navitem\\"><Link to=\\"/\\">Home</Link></li>
      <li className=\\"navitem\\"><Link to=\\"/calculator\\">Calculator</Link></li>
      <li className=\\"navitem\\"><Link to=\\"/quote\\">Quote</Link></li>
    </ul>
  </header>"
`;
